### DISCUSSION TITLE
Contextual Menus

### DISCUSSION THREAD

**Discussion by Moosems** [pyinstaller/pyinstaller#6829] [Help] [answered]
How would I create a contextual/shortcut menu (Right Click menu) and add commands such as Make new window, list old files opened, etc?

**Comment by rokm** [comment, marked as answer]
I *think* the menu items that are shown once the application is running are handled by the UI toolkit (provided the toolkit offers support for tray icon and menu there), so you'll have to look there. If it is possible to have static entries, then they are probably handled by `Info.plist` (assuming we're talking about macOS due to #6828). I think they are handled by the running application, though, because options like "New Window" and "New Private Window" go away from pinned Firefox icon when the application is closed...

But you're probably better off asking in some Apple and macOS forum, as this is OS-specific and is not directly related to PyInstaller.

**Reply by Moosems** [reply]
Do you have any idea how I might be able to do this with tkinter?

**Reply by rokm** [reply]
Nope, sorry.

**Reply by GlassGhost** [reply]
I don't know about tkinter but a QT library like Pyside should make it possible.

pyinstaller should be able to set that up for you, but you would have to start your code for your current gui from scratch;
that said you can create a ui file in qt creator and load it with the right function, which should expedite that process.

However Pyside doesn't have a bdist wheel target, and 

I made a simple python package manager:

https://github.com/GetWheel/tireshop

But I am having trouble building packages that don't have bdist_wheel targets.

Do you know anyone who would know how to write a bdist_wheel target for pyside?

it's a very large project, and I don't want to go into it by myself.

**Reply by bwoodsend** [reply]
`python setup.py bdist_wheel` is deprecated in favour of `pip wheel --no-deps .`. That said, `pyside` is also EOL in favour of `pyside2` and `pyside6`. Any reason you can't just use those since they come with wheels precompiled?

**Reply by GlassGhost** [reply]
The pyside2 github linked to pyside github saying it was newer, I thought pyside6 was for qt6.

Sorry I got the wrong command, will try everything. Appreciate the comment a lot.

I'm thinking about making a gui for managing the git versions/tags available for creating wheels on the tirerack, if I get pyside working.

I'm probably going to base it off a ui file from qt creator.

**Reply by bwoodsend** [reply]
What exactly is tireshop supposed to do that you can't already do via:
```bash
pip wheel numpy==1.19.3 pandas==1.4.1 sphinx==4.4.0 six==1.16.0 pyinstaller==4.10 "OpenGLContext @ git+https://github.com/mcfletch/openglcontext.git@dd03eba62c5636cff309e8def4422af807614288"
```
(or `pip wheel -r requirements.txt`)?

**Comment by Moosems** [comment]
@rokm, Certain applications still have a persistent memory for these menus or will show menus when closed (E.G. right click `System Preferences` when closed) which would indicate that, since there is no running process for the app to create such menus, the bundle itself can, in some way, influence the way these operate. The documentation for the NSMenu (the menu for Docks) can be found [here](https://developer.apple.com/documentation/appkit/nsmenu) and I hope this can be looked into a bit. If not, I totally understand. Even if it *is* possible, it would likely be a decent undertaking but, in my opinion, it is well worth the work. Let me know what you think.
